app/src/main/java/com/epam/themes/backend/StudentsWebService.java

```diff
@@ -7,44 +7,130 @@
-            student.setId((long) i);
-            student.setHwCount(1 + mRandom.nextInt(5));
+            student.setId(lastId);
+            student.setHwCount(1 + random.nextInt(5));
+            student.setName(nameStudents.get(i));
             student.setName(String.valueOf(i));
-            mStudents.add(student);
+            studentList.add(student);
         }
     }
 
     @Override
-    public void getEntities(ICallback<List<Student>> pCallback) {
+    public void getEntities(final ICallback<List<Student>> callback) {
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                callback.onResult(studentList);
+            }
+        }, 3000);
     }
 
     @Override
-    public void getEntities(final int pStartRange,
-                            final int pEndRange,
-                            final ICallback<List<Student>> pCallback) {
-        mHandler.postDelayed(new Runnable() {
+    public void getEntities(final int startRange,
+                            final int endRange,
+                            final ICallback<List<Student>> callback) {
+        handler.postDelayed(new Runnable() {
+
+            @Override
+            public void run() {
+                if (startRange < studentList.size()) {
+                    if (endRange > studentList.size()) {
+                        callback.onResult(studentList.subList(startRange, studentList.size()));
+                    } else {
+                        callback.onResult(studentList.subList(startRange, endRange));
+                    }
+                } else {
+                    callback.onResult(null);
+                }
+            }
+        }, 3000);
+    }
 
+    public void insertEntity(final int position, final Student student, final ICallback<Long> callback) {
+        handler.postDelayed(new Runnable() {
             @Override
             public void run() {
-                pCallback.onResult(mStudents.subList(pStartRange, pEndRange));
+                lastId++;
+                student.setId(lastId);
+                studentList.add(position, student);
+                callback.onResult(lastId);
             }
-        }, 1000);
+        }, 3000);
     }
 
     @Override
-    public void removeEntity(Long pId) {
+    public void removeEntity(final Long id) {
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                for (int i = 0; i < studentList.size(); i++) {
+                    if (studentList.get(i).getId().equals(id)) {
+                        studentList.remove(i);
 
+                        break;
+                    }
+                }
+            }
+        }, 3000);
+    }
+
+    @Override
+    public void setHomework(final Long id, final int hwCount) {
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                for (int i = 0; i < studentList.size(); i++) {
```

Comment:
I understand that this code so strange because you emulate long operation, but just tip for future, running this method executing in main thread is wrong